content-src/components/DiscoveryStreamComponents/DSFooter/DSFooter.jsx

```diff
@@ -0,0 +1,46 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { cardContextTypes } from "../../Card/types.js";
+import React from "react";
+
+const StatusMessage = ({ icon, fluentID }) => (
+  <div className="status-message">
+    <span
+      aria-haspopup="true"
+      className={`story-badge-icon icon icon-${icon}`}
+    />
+    <div className="story-context-label" data-l10n-id={fluentID} />
+  </div>
+);
+
+export class DSFooter extends React.PureComponent {
+  static getDerivedStateFromProps(nextProps, prevState) {
+    return {
+      hasUpdate: prevState && prevState.badge !== nextProps.badge,
+      ...nextProps,
+    };
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (this.state.badge !== prevState.badge) {
+      // eslint-disable-next-line react/no-did-update-set-state
+      this.setState({ hasUpdate: false });
```

Comment:
@mardak Here to display transition, there is a forced re-render by adding StatusMessage thats removed in initial render with hasUpdate as true inside getDerivedStateFromProps. This looks good to me, including you for React feedback and if it can be optimized further.

Reply:
What transition are you referring to? CSS animation should happen right away if you want something to start when rendered. From a quick test of selecting "Bookmark" from the menu then hitting escape. Also, seems like the message persists to the next new tab -- expected?

![Jul-24-2019 14-50-56](https://user-images.githubusercontent.com/438537/61831400-9432f780-ae22-11e9-8ae7-803e4bba91d6.gif)

Reply:
Also, with preloaded newtab, the message is already shown with no animation which may or may not be expected.

Reply:
My understanding animation is shown to catch attention of changing context message when user click on 'Bookmark' or 'Save to Pocket' in Context Menu. There is somework to add green color to text while animating that needs to be implemented.

In all other cases that includes preloaded tab or opening a new tab, message should persist and  should behave similar to 'Sponsored by ...' text.

Reply:
Just saw what you mean by " From a quick test of selecting "Bookmark" from the menu then hitting escape." experience captured in image above. Yeah that could be tricky. FWIW its parity and behave similar in AS :)

Reply:
To support not animating text in all other cases such as refresh and opening newtab  may be we should maintain [class animate](https://gist.github.com/punamdahiya/db6f603cff063bb977b778ce0388eab5) that should be applied to status-message when prevState inside getDerivedStateFromProps is not undefined.

Reply:
Sounds like a good idea. Though, using prevState as undefined will prevent the first animation load from not working (from no badge to badge). Adding an extra case 

```
    return {
      // Only time we dont animate is when our first initial render (prevState == null)
      // has a context_type cached from the same browser session.
      shouldNotAnimate: !prevState && nextProps.context_type,
      hasUpdate: prevState && prevState.context_type !== nextProps.context_type,
      context_type: nextProps && nextProps.context_type,
    };
```

Might be confusing to have hasUpdate and shouldNotAnimate but I need hasUpdate to toggle the component (in the case of "Bookmark" to "Remove bookmark" to refresh the animation) and shouldNotAnimate to toggle the class (in the case of opening a new tab with the "Bookmark" badge already there from a previous session)